File: templates/android/app/src/main/kotlin/{{PACKAGE_NAME}}/gyo/ConsoleBridgeHandler.kt
package {{PACKAGE_NAME}}.gyo

import android.util.Log
import org.json.JSONArray
import org.json.JSONObject

/**
 * Console Bridge Handler - handles console logs from web
 */
class ConsoleBridgeHandler : BridgeHandler {
    override fun handle(method: String, data: JSONObject): Any? {
        when (method) {
            "log" -> {
                val level = data.optString("level", "log")
                val args = data.optJSONArray("args") ?: JSONArray()
                val timestamp = data.optLong("timestamp", System.currentTimeMillis())
                
                val message = formatConsoleMessage(args)
                val tag = "WebConsole"
                
                when (level) {
                    "error" -> Log.e(tag, message)
                    "warn" -> Log.w(tag, message)
                    "info" -> Log.i(tag, message)
                    "debug" -> Log.d(tag, message)
                    else -> Log.d(tag, message)
                }
                
                return mapOf("success" to true)
            }
            else -> throw IllegalArgumentException("Unknown console method: $method")
        }
    }
    
    private fun formatConsoleMessage(args: JSONArray): String {
        val parts = mutableListOf<String>()
        for (i in 0 until args.length()) {
            val arg = args.get(i)
            parts.add(formatArg(arg))
        }
        return parts.joinToString(" ")
    }
    
    private fun formatArg(arg: Any?): String {
        return when (arg) {
            is JSONObject -> {
                if (arg.has("__type")) {
                    when (arg.getString("__type")) {
                        "Error" -> "${arg.optString("name")}: ${arg.optString("message")}"
                        "Date" -> arg.optString("value")
                        else -> arg.toString()
                    }
                } else {
                    arg.toString()
                }
            }
            is JSONArray -> arg.toString()
            null -> "null"
            else -> arg.toString()
        }
    }
}

File: templates/ios/Sources/Gyo/ConsoleBridgeHandler.swift
import Foundation

// MARK: - Console Bridge Handler

final class ConsoleBridgeHandler: BridgeHandler {
    func handle(method: String, data: [String: Any]) throws -> Any? {
        switch method {
        case "log":
            let level = data["level"] as? String ?? "log"
            let args = data["args"] as? [[String: Any]] ?? []
            let _ = data["timestamp"] as? Int64 ?? 0  // timestamp not used yet
            
            let message = formatConsoleMessage(args)
            
            // Use print for cross-platform compatibility
            print("[\(level.uppercased())] \(message)")
            
            return ["success": true]
            
        default:
            throw BridgeError.unknownMethod(method)
        }
    }
    
    private func formatConsoleMessage(_ args: [[String: Any]]) -> String {
        let parts = args.map { formatArg($0) }
        return parts.joined(separator: " ")
    }
    
    private func formatArg(_ arg: [String: Any]) -> String {
        if let type = arg["__type"] as? String {
            switch type {
            case "Error":
                let name = arg["name"] as? String ?? "Error"
                let message = arg["message"] as? String ?? ""
                return "\(name): \(message)"
            case "Date":
                return arg["value"] as? String ?? ""
            default:
                return String(describing: arg)
            }
        }
        
        // Try to extract the actual value from the dictionary
        if let value = arg["value"] {
            return String(describing: value)
        }
        
        return String(describing: arg)
    }
}
